src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -25,9 +39,111 @@ public FindCommand parse(String args) throws ParseException {
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        String[] keywords;
+        if (args.contains(PREFIX_NAME.getPrefix())) {
+            keywords = parseName(args).split("\\s+");
+            AttributeContainsKeywordsPredicate type = new NameContainsKeywordsPredicate(Arrays.asList(keywords));
+            return new FindCommand(type);
+        } else if (args.contains(PREFIX_PHONE.getPrefix())) {
+            keywords = parsePhone(args).split("\\s+");
+            AttributeContainsKeywordsPredicate type = new PhoneContainsKeywordsPredicate(Arrays.asList(keywords));
+            return new FindCommand(type);
+        } else if (args.contains(PREFIX_EMAIL.getPrefix())) {
+            keywords = parseEmail(args).split("\\s+");
+            AttributeContainsKeywordsPredicate type = new EmailContainsKeywordsPredicate(Arrays.asList(keywords));
+            return new FindCommand(type);
+        } else if (args.contains(PREFIX_ACADEMIC_MAJOR.getPrefix())) {
+            keywords = parseAcademicMajor(args).split("\\s+");
+            AttributeContainsKeywordsPredicate type =
+                    new AcademicMajorContainsKeywordsPredicate(Arrays.asList(keywords));
+            return new FindCommand(type);
+        } else {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+    }
+
+    /**
+     * Parses the given {@code String} of arguments into a {@code String name}
+     * which will be used for {@code FindCommandParser} object for execution.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the user input does not conform to {@code Name} format.
+     */
+    public String parseName(String args) throws ParseException {
```

Comment:
Actually I think all these methods are redundant 🤔 . We already have all these similar methods in the `ParserUtil.java` just that those methods return attributes object instead of String. I think we can just use parse methods, then use `.toString()` to change them to String value.

Example:
```java
if (args.contains(PREFIX_NAME.getPrefix())) {
            keywords = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get()).toString().split("\\s+");
            AttributeContainsKeywordsPredicate type = new NameContainsKeywordsPredicate(Arrays.asList(keywords));
            return new FindCommand(type);
```

What we need to do is to introduce an `ArgumentMultimap` similar to how it is done in the `AddCommandParser`

Example:
```java
 ArgumentMultimap argMultimap =
                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE,
                        PREFIX_EMAIL, PREFIX_ACADEMIC_MAJOR, PREFIX_TAG);
```

We then wouldn't need all these parser methods.